editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/globalconfiguration/beans/BeanConfigUtil.java

```diff
@@ -526,7 +527,8 @@ public void editBeanArgument(Element xmlElement, String type, String value) {
 	}
 	
 	public Element createBeanNode(final CamelFile camelFile, String id, String className) {
-		final String prefixNS = camelFile.getRouteContainer().getXmlNode().getPrefix();
+		// get NS prefix from parent document, not route container node
+		final String prefixNS = camelFile.getRouteContainer().getXmlNode().getOwnerDocument().getPrefix();
```

Comment:
seems that it should be available from https://docs.oracle.com/javase/7/docs/api/javax/xml/namespace/NamespaceContext.html but don't know how to get it.

otherwise, at least, it would be nice to get the "beans" node (in the correct namespace uri) and retrieve the prefix from here instead of the route one which we don't know what it is for sure

and a last thing is that it seems that there is a specific method to create the element with a specific namespace (which might explain the "uncategorized" place when creating a new Bean in global bean editor if it is not used): https://docs.oracle.com/javase/7/docs/api/org/w3c/dom/Document.html#createElementNS(java.lang.String,%20java.lang.String)
but still requires to find the correct prefix